inc/elasticsearch_packages/namespace.php

```diff
@@ -0,0 +1,432 @@
+					'DomainName' => get_elasticsearch_domain(), // required.
+					'PackageID' => $real_package_id, // required.
+				] );
+				$package = $result['DomainPackageDetails'];
+				$status = $package['DomainPackageStatus'];
+			}
+		} else {
+			// Package is currently being associated or is active.
+			$package = $packages['DomainPackageDetailsList'][0];
+			$status = $package['DomainPackageStatus'];
+		}
+
+		// Update the stored status.
+		if ( $for_network ) {
+			update_site_option( "altis_search_package_status_{$slug}", $status );
+			update_site_option( "altis_search_package_error_{$slug}", $package['ErrorDetails']['ErrorMessage'] ?? null );
+		} else {
+			update_option( "altis_search_package_status_{$slug}", $status );
+			update_option( "altis_search_package_error_{$slug}", $package['ErrorDetails']['ErrorMessage'] ?? null );
+		}
+
+		// If the package is now active then update the index settings.
+		if ( $status === 'ACTIVE' ) {
+			// Add the existing package ID to list of packages to remove.
+			// These packages are removed after the index settings are updated.
+			add_package_to_remove( $existing_package_id );
+
+			// Unschedule this status check.
+			$next = wp_next_scheduled( 'altis.search.check_package_status', $check_status_hook_args );
+			if ( $next ) {
+				wp_unschedule_event( $next, 'altis.search.check_package_status', $check_status_hook_args );
+			}
+
+			// Schedule the index settings update.
+			if ( ! wp_next_scheduled( 'altis.search.update_index_settings', $update_index_hook_args ) ) {
+				wp_schedule_single_event( time() + MINUTE_IN_SECONDS, 'altis.search.update_index_settings', $update_index_hook_args );
+			}
+			return;
+		}
+
+		// Queue up another check if we're still copying, associating or validating.
+		$recheck_statuses = [
+			'COPYING',
+			'ASSOCIATING',
+			'VALIDATING',
+		];
+		if ( ! in_array( $status, $recheck_statuses, true ) ) {
+			$error_message = sprintf( 'Package %s has encountered an error. See the status for more details.', $package_id );
+			throw new Exception( $error_message );
+		}
+	} catch ( Exception $e ) {
+		// Unschedule this hook.
+		$next = wp_next_scheduled( 'altis.search.check_package_status', $check_status_hook_args );
+		if ( $next ) {
+			wp_unschedule_event( $next, 'altis.search.check_package_status', $check_status_hook_args );
+		}
+
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		trigger_error( $e->getMessage(), E_USER_WARNING );
+	}
+}
+
+/**
+ * Loop over packages to remove and delete them. Called on
+ * the altis.search.updated_all_index_settings hook.
+ *
+ * @return void
+ */
+function on_update_index_settings() {
+	$packages_to_remove = get_packages_to_remove();
+
+	foreach ( $packages_to_remove as $package_id ) {
+		delete_package( $package_id );
+	}
+
+	// Empty the array of packages to be removed.
+	update_site_option( 'altis_search_packages_to_remove', [] );
+}
+
+/**
+ * Hook triggered when deleting a package directly.
```

Comment:
Switch to imperative(?):
```suggestion
 * Handle direct package deletion.
```